tests/test_glm.py

```diff
@@ -2626,6 +2629,49 @@ def test_feature_mask_compatibility_fit(
             else:
                 getattr(model, attr_name)(X, y)
 
+    @pytest.mark.parametrize(
+        "reg_setup",
+        [
+            "population_poissonGLM_model_instantiation",
+            "population_poissonGLM_model_instantiation_pytree",
+        ],
+    )
+    def test_metadata_pynapple_fit(self, reg_setup, request):
+        X, y, model, true_params, firing_rate = request.getfixturevalue(reg_setup)
+        y = TsdFrame(
+            t=np.arange(y.shape[0]), d=y, metadata={"y": np.arange(y.shape[1])}
+        )
+        model.fit(X, y)
+        assert hasattr(model, "_metadata") and (model._metadata is not None)
+        assert np.all(y.metadata == model._metadata["metadata"])
```

Comment:
```suggestion
        assert np.all(y._metadata == model._metadata["metadata"])
```
if you make the change above, then this will need to get changed too